application/src/userlistmodel.cpp

```diff
@@ -0,0 +1,81 @@
+#include "userlistmodel.h"
+#include "databaseengine.h"
+
+UserListModel::UserListModel(QObject *parent) : QAbstractListModel(parent)
+{
+    connectDatabase();
+}
+
+void UserListModel::addUserlist(QList<User> const &userList)
+{
+
+    int size = m_userList.size();
+
+    if (size != userList.size()) {
+        beginInsertRows(QModelIndex(), size, userList.size()-1);
+
+        for (int i = size; i <= userList.size() - 1; i++) {
+            m_userList.append(userList[i]);
+        }
```

Comment:
Мы ведь добавляем весь список в конец, который передаётся в аргумент функции. То есть есть достаточно так:
```suggestion
        m_userList.append(userList);
```